winlogbeat/eventlog/wineventlog_experimental.go

```diff
@@ -241,45 +242,60 @@ func (l *winEventLogExp) Close() error {
 // newWinEventLogExp creates and returns a new EventLog for reading event logs
 // using the Windows Event Log.
 func newWinEventLogExp(options *common.Config) (EventLog, error) {
+	var xmlQuery string
+	var err error
+	var isFile bool
+	var log *logp.Logger
+
 	cfgwarn.Experimental("The %s event log reader is experimental.", winEventLogExpAPIName)
 
 	c := winEventLogConfig{BatchReadSize: 512}
 	if err := readConfig(options, &c); err != nil {
 		return nil, err
 	}
 
-	queryLog := c.Name
-	isFile := false
-	if info, err := os.Stat(c.Name); err == nil && info.Mode().IsRegular() {
-		path, err := filepath.Abs(c.Name)
+	id := c.ID
+	if c.XMLQuery != "" {
+		xmlQuery = c.XMLQuery
+		log = logp.NewLogger("wineventlog").With("id", id)
+	} else {
+		if id == "" {
+			id = c.Name
+		}
+
+		queryLog := c.Name
+		if info, err := os.Stat(c.Name); err == nil && info.Mode().IsRegular() {
```

Comment:
I know that this is pre-existing, but is it ever likely that a user will specify a file log destination with a file scheme?

Reply:
I'm not entirely sure, perhaps someone else can elaborate on that more (@andrewkroh ?). We do support providing a filename and reading from it, that much I do know.

Reply:
In the docs we say they need to pass an absolute path so I don't think users will pass a file URI.

> To read events from an archived .evtx file you can specify the name as the absolute path (it cannot be relative) to the file.

https://www.elastic.co/guide/en/beats/winlogbeat/7.16/configuration-winlogbeat-options.html#configuration-winlogbeat-options-event_logs-name